legistar/events.py

```diff
@@ -182,32 +199,45 @@ def rollcalls(self, event):
                 for item in response.json():
                     yield item
 
+    def web_results(self, event):
+        api_key = (event['EventBodyName'].strip(),
+                   event['start'])
+
+        if api_key in self._scraped_events:
+            return self._scraped_events[api_key]
+
+        else:
+            for event, web_scraper in self._events:
+                web_key = self._event_key(event, web_scraper)
+                self._scraped_events[web_key] = event
+                if web_key == api_key:
+                    return event
+
     def _scrapeWebCalendar(self):
+        '''Generator yielding events from Legistar in roughly reverse
+        chronological order.
+        '''
         web_scraper = LegistarEventsScraper(self.jurisdiction,
                                             self.datadir,
                                             strict_validation=self.strict_validation,
                                             fastmode=(self.requests_per_minute == 0))
+
         web_scraper.EVENTSPAGE = self.EVENTSPAGE
         web_scraper.BASE_URL = self.WEB_URL
         web_scraper.TIMEZONE = self.TIMEZONE
         web_scraper.date_format = '%m/%d/%Y'
 
-        web_info = {}
-
-        for event, _ in web_scraper.events(follow_links=False):
-            key = self._event_key(event, web_scraper)
-            web_info[key] = event
-
-        return web_info
+        for year in reversed(range(web_scraper.now().year + 1)):
+            for event, _ in web_scraper.events(follow_links=False, since=year):
```

Comment:
Let's move this `year` iteration logic into web_scraper.events.

It can be like

```python
if since_year is None:
     since_year = 2000

for year in range(web.scraper.now().year, since_year - 1, -1):
     iterate the thing
```

we also need to make sure we do something sensical when `year` is not a valid value.

Reply:
Is 2000 a sane floor for all the scrapers?

Reply:
no, set the floor to 0 and figure out how to handle invalid values.